samples/bottom_navigation_bar/src/sample/bottom_navigation_bar.cljd

```diff
@@ -0,0 +1,39 @@
+(ns sample.bottom-navigation-bar
+  "Faithful port of https://docs.flutter.dev/cookbook/design/tabs"
+  (:require
+   ["package:flutter/material.dart" :as m]
+   [cljd.flutter :as f]))
+
+(defn main
+  []
+  (let
+      [title "Bottom Navigation Bar Demo"
+       pages [(m/Icon. m/Icons.directions_car)
+              (m/Icon. m/Icons.directions_transit)
+              (m/Icon. m/Icons.directions_bike)]
+       selected-index (atom 0)]
+    (f/run
+      (m/MaterialApp .title title)
+
+      .home
+      (m/Scaffold
+
+       .appBar
+       (m/AppBar
+        .title
+        (m/Text "BottomNavigationBar"))
+       
+       .body
+       (m/Center
+        .child
+        (f/widget :watch [current-index selected-index] (get pages current-index)))
```

Comment:
you can replace the whole form with, it will expand the same (f/widget macro & f/run allow that kind of threading)
```
.body
m/Center
:watch [current-index selected-index] (get pages current-index))
```